magento/pwa-studio
# [PWA-2879] Custom Beacon extension event

## Description

As a Merchant, I want to be able to track custom, Beacon-specific events on my storefront so that I may track events not dispatched by the default eventing framework. Beacon extension extends event framework with a searchResponseReceived event.

## Related Issue







Closes [PWA-2879](https://jira.corp.magento.com/browse/PWA-2879)

## Acceptance





### Verification Stakeholders


@dpatil-magento 

### Specification




## Verification Steps






#### Test scenario(s) for direct fix/feature




1. Search for a product in the search bar
2. Open browser console and verify that the sample extension logged the search response received event (with its contextual data)

#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
